File: dotnet-backend/projekatFlutter/Models/TripWaypoint.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace projekatFlutter.Models
{
    public class TripWaypoint:IEntity<int>
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int TripId { get; set; }

        [Required]
        public int SequenceNumber { get; set; }

        [Required]
        [MaxLength(100)]
        public string WaypointName { get; set; } = string.Empty;

        [MaxLength(255)]
        public string? Address { get; set; }

        [Required]
        [Column(TypeName = "decimal(10,8)")]
        public decimal Latitude { get; set; }

        [Required]
        [Column(TypeName = "decimal(11,8)")]
        public decimal Longitude { get; set; }

        public DateTime? PlannedArrivalTime { get; set; }
        public DateTime? ActualArrivalTime { get; set; }
        public DateTime? PlannedDepartureTime { get; set; }
        public DateTime? ActualDepartureTime { get; set; }

        [Required]
        [MaxLength(30)]
        public string Status { get; set; } = "Pending";

        [MaxLength(500)]
        public string? Notes { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation Properties
        [ForeignKey("TripId")]
        public Trip Trip { get; set; } = null!;
    }
}

File: frontend/lib/models/trip_waypoint.dart
class TripWaypoint {
  final int id;
  final int tripId;
  final int sequenceNumber;
  final String waypointName;
  final String? address;
  final double latitude;
  final double longitude;
  final DateTime? plannedArrivalTime;
  final DateTime? actualArrivalTime;
  final DateTime? plannedDepartureTime;
  final DateTime? actualDepartureTime;
  final String status;
  final String? notes;
  final DateTime createdAt;

  TripWaypoint({
    required this.id,
    required this.tripId,
    required this.sequenceNumber,
    required this.waypointName,
    this.address,
    required this.latitude,
    required this.longitude,
    this.plannedArrivalTime,
    this.actualArrivalTime,
    this.plannedDepartureTime,
    this.actualDepartureTime,
    required this.status,
    this.notes,
    required this.createdAt,
  });

  factory TripWaypoint.fromJson(Map<String, dynamic> json) {
    return TripWaypoint(
      id: json['id'] as int,
      tripId: json['tripId'] as int,
      sequenceNumber: json['sequenceNumber'] as int,
      waypointName: json['waypointName'] as String,
      address: json['address'] as String?,
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      plannedArrivalTime: json['plannedArrivalTime'] != null
          ? DateTime.parse(json['plannedArrivalTime'] as String)
          : null,
      actualArrivalTime: json['actualArrivalTime'] != null
          ? DateTime.parse(json['actualArrivalTime'] as String)
          : null,
      plannedDepartureTime: json['plannedDepartureTime'] != null
          ? DateTime.parse(json['plannedDepartureTime'] as String)
          : null,
      actualDepartureTime: json['actualDepartureTime'] != null
          ? DateTime.parse(json['actualDepartureTime'] as String)
          : null,
      status: json['status'] as String,
      notes: json['notes'] as String?,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'tripId': tripId,
      'sequenceNumber': sequenceNumber,
      'waypointName': waypointName,
      'address': address,
      'latitude': latitude,
      'longitude': longitude,
      'plannedArrivalTime': plannedArrivalTime?.toIso8601String(),
      'actualArrivalTime': actualArrivalTime?.toIso8601String(),
      'plannedDepartureTime': plannedDepartureTime?.toIso8601String(),
      'actualDepartureTime': actualDepartureTime?.toIso8601String(),
      'status': status,
      'notes': notes,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}